src/Scanner.h

```diff
@@ -0,0 +1,55 @@
+#ifndef SCANNER_H
+#define SCANNER_H
+
+#include <iostream>
+#include <string>
+#include <list>
+#include <map>
+#include "Token.h"
+
+
+constexpr char NULLITERAL[] = " ";
+namespace loxs {
+    class Scanner {
+        private:
+            std::string source; //Code source
+            std::list<Token> tokens; //List of tokens
+            void scanToken();
+            bool end_of_file();
+            char peek();
+            void string_literal();
+            void number_literal();
+            void identify_literal();
+            bool match(char expected);
+            char next_char();
+            long unsigned current = 0; //Points the current char of lexeme
+            long unsigned start = 0;  //Points to the first char of lexem
+            long unsigned line = 1;  //Points to the line where current is
```

Comment:
these can be `std::size_t`?

Reply:
`std::size_t` isn't better for uses cases like loops and index ? (https://en.cppreference.com/w/cpp/types/size_t) in the other hand, we are using `start` and `current `to fetch sub strings from a `std::string`